client/src/store/notification/notification.slice.ts

```diff
@@ -0,0 +1,121 @@
+import { apiSlice } from '../api';
+import { ISendNotificationRequestBody, ISubscription } from '@/types';
+
+const notification = apiSlice.injectEndpoints({
```

Comment:
Since the saveSubscription and unsubscribe endpoints change user data, we might want to invalidate those tags. See user.slice

This is still something I'm not entirely familiar with and for further reading you can see this: https://redux-toolkit.js.org/rtk-query/usage/automated-refetching

Reply:
I think for the sake of time we should take this into consideration after we've reached our MVP and can start optimizing the project. This is a good catch, though. Thank you for bringing it to my attention.